todo_app/oauth.py

```diff
@@ -9,14 +9,18 @@
 )
 
 from functools import wraps
-from flask import g, request, redirect, url_for
+from flask import Flask, g, request, redirect, url_for
 
 
-def login_required(f):
-    @wraps(f)
-    def decorated_function(*args, **kwargs):
-        if not github.authorized:
-            return redirect(url_for("github.login"))
-        return f(*args, **kwargs)
+def login_required(app: Flask):
```

Comment:
Probably easier to use `from flask import current_app` here rather than add an extra variable to `login_required`. Similarly for `MongoItemsService` & `BaseItemsService`.